tests/compile/large_integer_T5290.py

```diff
@@ -0,0 +1,4 @@
+# mode: compile
```

Comment:
Why not make this a runnable tests? It seems worth testing that the number passes correctly and that we only shorten the cnames (we can probably test that with a tree assertion, looking at the cname attribute of the constant).

It also seems worth testing that the uniqueness algorithm works. This is not currently covered here.